tidepool_data_science_metrics/cgm/cgm.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+from typing import Tuple
+
+import tidepool_data_science_metrics.common.common as common
+
+
+def cv_of_glucose(bg_values, round_val=2):
+    """
+    Calculate the coefficient of variation on set of glucose values
+
+    Parameters
+    ----------
+    bg_values : ndarray
+        1D array containing data with `int` type.
+    round_val : int
+        The number of digits to round the result to.
+
+    Returns
+    -------
+    int
+        The calculated Coefficient of variation
+    """
+    std_dev = common.std_deviation(bg_values, round_val)
+    avg_glu = common.avg(bg_values, round_val)
+    return round(std_dev / avg_glu * 100, round_val)
+
+
+def gmi(bg_values, round_val=2):
+    """
+    Calculate the Glucose Management Indicator on set of glucose values. GMI indicates the average
+    A1C level that would be expected based on mean glucose measured
+
+    Parameters
+    ----------
+    bg_values : ndarray
+        1D array containing data with `int` type.
+    round_val : int
+        The number of digits to round the result to.
+
+    Returns
+    -------
+    int
+        The calculated Glucose Management Indicator
+    """
+    gmi = 3.31 + (0.02392 * common.mean(bg_values))
+    return round(gmi, round_val)
+
+
+def percent_values_by_range(
```

Comment:
I like the generalization of percent_values_by_range, and also think we should add in specific and common ranges (e.g., percent_values_in_range_70-180, percent_values_below_<>, where <> can be 40, 54, 70. 

A great test to add would be to calculate % values < 70, between 70-180, and > 180 and then to make sure that the total of all 3 percentages added up to 100%.

Reply:
here is a bunch that we can add immediately % Time in target range (70 – 180 mg/dL)
% Time > 180 mg/dL
Mean glucose
% Time < 70 mg/dL
% Time < 54 mg/dL
% Time > 250

Reply:
when you say specific and common ranges...do you mean have it auto generate those or have different functions for each ?

Reply:
added the functions